code/modules/mob/dead/observer/observer.dm

```diff
@@ -570,44 +572,9 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 
 	var/datum/gas_mixture/environment = usr.loc.return_air()
 
-	var/pressure = environment.return_pressure()
-	var/total_moles = environment.total_moles()
-
-	src << "<span class='notice'><B>Results:</B></span>"
-	if(abs(pressure - ONE_ATMOSPHERE) < 10)
-		src << "<span class='notice'>Pressure: [round(pressure,0.1)] kPa</span>"
-	else
-		src << "<span class='warning'>Pressure: [round(pressure,0.1)] kPa</span>"
-	if(total_moles)
-		var/o2_concentration = environment.oxygen/total_moles
-		var/n2_concentration = environment.nitrogen/total_moles
-		var/co2_concentration = environment.carbon_dioxide/total_moles
-		var/plasma_concentration = environment.toxins/total_moles
-
-		var/unknown_concentration =  1-(o2_concentration+n2_concentration+co2_concentration+plasma_concentration)
-		if(abs(n2_concentration - N2STANDARD) < 20)
-			src << "<span class='notice'>Nitrogen: [round(n2_concentration*100)]% ([round(environment.nitrogen,0.01)] moles)</span>"
-		else
-			src << "<span class='warning'>Nitrogen: [round(n2_concentration*100)]% ([round(environment.nitrogen,0.01)] moles)</span>"
-
-		if(abs(o2_concentration - O2STANDARD) < 2)
-			src << "<span class='notice'>Oxygen: [round(o2_concentration*100)]% ([round(environment.oxygen,0.01)] moles)</span>"
-		else
-			src << "<span class='warning'>Oxygen: [round(o2_concentration*100)]% ([round(environment.oxygen,0.01)] moles)</span>"
-
-		if(co2_concentration > 0.01)
-			src << "<span class='warning'>CO2: [round(co2_concentration*100)]% ([round(environment.carbon_dioxide,0.01)] moles)</span>"
-		else
-			src << "<span class='notice'>CO2: [round(co2_concentration*100)]% ([round(environment.carbon_dioxide,0.01)] moles)</span>"
-
-		if(plasma_concentration > 0.01)
-			src << "<span class='warning'>Plasma: [round(plasma_concentration*100)]% ([round(environment.toxins,0.01)] moles)</span>"
-
-		if(unknown_concentration > 0.01)
-			src << "<span class='warning'>Unknown: [round(unknown_concentration*100)]% ([round(unknown_concentration*total_moles,0.01)] moles)</span>"
-
-		src << "<span class='notice'>Temperature: [round(environment.temperature-T0C,0.1)]&deg;C</span>"
-		src << "<span class='notice'>Heat Capacity: [round(environment.heat_capacity(),0.1)]</span>"
+	if(!analyzer)
+		analyzer = new(usr)
+	usr << analyzer.output_gas_scan(environment, usr.loc, 1)

```

Comment:
But isn't it checked by istype above this?
istype returns 0 for no datum, right?